### PR TITLE
fix #11845 contains and limit bug for relationship field

### LINKED ISSUE #11845: `_limit` passed to request to fetch options of relationship type field (select box) should be `limit`
### Describe the bug

More records than intended are being fetch for the select box of the relationship field type. This causes the page to crash.

### Steps to reproduce the behavior

1. Create a relationship field
2. Create more than 20 records in the relationship table
3. Go to the record in which you can link records from the relationship table.
4. Inspect the request being done. 
5. See that more than 20 records are being returned despite the `_limit` param being passed to the request.

### Expected behavior

Only 20 records should be returned. (I currently have 30.000 records in which I want to search, this causes the page to break)

### Screenshots

*With `_limit` (wrong)*
Screenshot 2021-12-08 at 12 18 43

*With `limit` (correct)*
Screenshot 2021-12-08 at 12 18 27

### Code snippets

N/A

### System

- Node.js version: 14
- Strapi version: v4
- Database: mysql

### Additional context

I currently don't have a chance of creating a PR given my available time.

Probable responsible line of code: https://github.com/strapi/strapi/blob/0f9b69298b2d94b31b434bd7217060570ae89374/packages/core/admin/admin/src/content-manager/components/SelectWrapper/index.js

### PR DESCRIPTION
**Pull request by sublayerio**
**Note:** This PR isn't tested.

See #11845 for more info.

### What does it do?

changes

```js
const params = { _limit: state._limit, ...defaultParams };

if (state._contains) {
  params[containsKey] = state._contains;
}
```

to

```js
const params = { limit: state._limit, ...defaultParams };

if (state._contains) {
  const [fieldName] = containsKey.split('_')
  params[`filters[${fieldName}][$contains]`] = state._contains;
}
```

### Why is it needed?

Page crashes when fetching large amount of related records.

### How to test it?

1. Create a relationship field
2. Create more than 20 records in the relationship table
3. Go to the record in which you can link records from the relationship table.
4. Inspect the request being done.
5. See that more than 20 records are being returned despite the _limit param being passed to the request.

### Related issue(s)/PR(s)

#11845

### CODE REVIEW

**Comment by derrickmehaffy** [comment]
@sublayerio can you please sign the CLA

**Comment by sublayerio** [comment]
@derrickmehaffy There you go

File: packages/core/admin/admin/src/content-manager/components/SelectWrapper/index.js [hunk 1]
```diff
@@ -141,10 +141,11 @@ function SelectWrapper({
 
       setIsLoading(true);
 
-      const params = { _limit: state._limit, ...defaultParams };
+      const params = { limit: state._limit, ...defaultParams };
 
       if (state._contains) {
-        params[containsKey] = state._contains;
+        const [fieldName] = containsKey.split('_')
```
Comment (soupette) [inline packages/core/admin/admin/src/content-manager/components/SelectWrapper/index.js:147]: Nice try! However this won't work if the name of the field contains more than one underscore.


In the content manager the layout handles all the query params, you can take a look at this [PR](https://github.com/strapi/strapi/pull/11898) which fixes the bug.

**Comment by soupette** [comment]
Thank you for your contribution, I am closing this PR as the provided code does not fix the issue.

**Pull request closed without merging**